src/corehost/cli/deps_resolver.cpp

```diff
@@ -403,6 +417,69 @@ void deps_resolver_t::init_known_entry_path(const deps_entry_t& entry, const pal
     }
 }
 
+void deps_resolver_t::resolve_additional_deps(const hostpolicy_init_t& init)
+{
+    pal::string_t additional_deps_serialized = init.additional_deps_serialized;
+    pal::string_t fx_name = init.fx_name;
+    pal::string_t fx_ver = init.fx_ver;
+
+    if (additional_deps_serialized.empty())
+    {
+        return;
+    }
+
+    pal::string_t additional_deps_path;
+    pal::stringstream_t ss(additional_deps_serialized);
+
+    // Process the delimiter separated custom deps files.
+    while (std::getline(ss, additional_deps_path, PATH_SEPARATOR))
+    {
+        // If it's a single deps file, insert it in 'm_additional_deps_files'
+        if (ends_with(additional_deps_path, _X(".deps.json"), false))
+        {
+            if (pal::file_exists(additional_deps_path))
+            {
+                m_additional_deps_files.push_back(additional_deps_path);
+            }
+        }
+        // If it's not a single deps file, we'll assume it's a directory
+        else
+        {
+            // We'll search deps files in 'base_dir'/shared/fx_name/fx_ver
+            append_path(&additional_deps_path, _X("shared"));
+            append_path(&additional_deps_path, fx_name.c_str());
```

Comment:
For standalone apps, as you said, there is no fx_ver/fx_name that is available since that information comes from runtimeconfig.json that does not have it for standalone publish. 

Also, since standalone apps already published in context of a given FX/Ver/RID, it is reasonable to expect the user to know that information and point explicitly to corresponding json files.